File: app/src/main/java/cn/com/woong/readhub/db/DBManager.java
package cn.com.woong.readhub.db;

import android.content.Context;

import java.util.ArrayList;
import java.util.List;

import cn.com.woong.readhub.bean.DaoMaster;
import cn.com.woong.readhub.bean.DaoSession;
import cn.com.woong.readhub.bean.NewsMo;
import cn.com.woong.readhub.bean.NewsMoDao;
import cn.com.woong.readhub.bean.TopicMo;
import cn.com.woong.readhub.bean.TopicMoDao;

public class DBManager {
    private Context mContext;
    private DaoMaster mDaoMaster;
    private DaoSession mDaoSession;
    private TopicMoDao mTopicMoDao;
    private NewsMoDao mNewsMoDao;

    private DBManager(Context context) {
        mContext = context;
    }

    private static volatile DBManager instance = null;
    public static DBManager getInstance(Context context){
        if (instance==null){
            synchronized (DBManager.class){
                if (instance==null){
                    instance = new DBManager(context);
                }
            }
        }
        return instance;
    }

    public void init() {
        DaoMaster.DevOpenHelper helper = new DaoMaster.DevOpenHelper(mContext, "readhub.db");
        mDaoMaster = new DaoMaster(helper.getWritableDb());
        mDaoSession = mDaoMaster.newSession();

        mTopicMoDao = mDaoSession.getTopicMoDao();
        mNewsMoDao = mDaoSession.getNewsMoDao();
    }

    public void insertTopicMo(TopicMo topicMo) {
        mTopicMoDao.insertOrReplace(topicMo);
    }

    public void insertNewsMo(NewsMo newsMo) {
        mNewsMoDao.insertOrReplace(newsMo);
    }

    public void deleteTopicMo(TopicMo topicMo) {
        mTopicMoDao.delete(topicMo);
    }

    public void deleteNewsMo(NewsMo newsMo) {
        mNewsMoDao.delete(newsMo);
    }

    public List<TopicMo> queryAllTopicMo() {
        return mTopicMoDao.queryBuilder().build().list();
    }

    public List<NewsMo> queryAllNewsMo() {
        return mNewsMoDao.queryBuilder().build().list();
    }
}

File: app/src/main/java/cn/com/woong/readhub/db/DBManager.kt
package cn.com.woong.readhub.db

import android.content.Context
import cn.com.woong.readhub.bean.NewsMo
import cn.com.woong.readhub.bean.TopicMo
import java.nio.file.Files.delete


class DBManager private constructor(private val mContext: Context) {
//    private var mDaoMaster: DaoMaster? = null
//    private var mDaoSession: DaoSession? = null
//    private var mTopicMoDao: TopicMoDao? = null
//    private var mNewsMoDao: NewsMoDao? = null
//
//    fun init() {
//        val helper = DaoMaster.DevOpenHelper(mContext, "readhub.db")
//        mDaoMaster = DaoMaster(helper.getWritableDb())
//        mDaoSession = mDaoMaster!!.newSession()
//
//        mTopicMoDao = mDaoSession!!.getTopicMoDao()
//        mNewsMoDao = mDaoSession!!.getNewsMoDao()
//    }
//
//    fun insertTopicMo(topicMo: TopicMo) {
//        mTopicMoDao!!.insertOrReplace(topicMo)
//    }
//
//    fun insertNewsMo(newsMo: NewsMo) {
//        mNewsMoDao!!.insertOrReplace(newsMo)
//    }
//
//    fun deleteTopicMo(topicMo: TopicMo) {
//        mTopicMoDao!!.delete(topicMo)
//    }
//
//    fun deleteNewsMo(newsMo: NewsMo) {
//        mNewsMoDao!!.delete(newsMo)
//    }
//
//    fun queryAllTopicMo(): List<TopicMo> {
//        return mTopicMoDao!!.queryBuilder().build().list()
//    }
//
//    fun queryAllNewsMo(): List<NewsMo> {
//        return mNewsMoDao!!.queryBuilder().build().list()
//    }
//
//    companion object {
//
//        @Volatile
//        private var instance: DBManager? = null
//
//        fun getInstance(context: Context): DBManager? {
//            if (instance == null) {
//                synchronized(DBManager::class.java) {
//                    if (instance == null) {
//                        instance = DBManager(context)
//                    }
//                }
//            }
//            return instance
//        }
//    }
}